lib/utilities.js

```diff
@@ -334,6 +337,33 @@ const jsonParse = (buffer) => {
   }
 };
 
+const flatObject = (sourceObject, fieldNames) => {
+  const targetObj = {};
+
+  for (const [key, value] of Object.entries(sourceObject)) {
+    if (!isObject(value)) {
+      targetObj[key] = value;
+      continue;
+    }
+
+    if (fieldNames.length > 0 && !fieldNames.includes(key)) {
+      targetObj[key] = { ...value };
+      continue;
+    }
+
+    for (const [key2, value2] of Object.entries(value)) {
+      const newKey = `${key}${toUpperCamel(key2)}`;
+
+      if (sourceObject[newKey] !== undefined)
+        throw new Error('flatObject: duplicate key');
+
+      targetObj[newKey] = value2;
+    }
+  }
+
+  return targetObj;
+};
```

Comment:
I suggest slightly naming improvement and default value usage for `fieldNames` argument
```js
const flatObject = (sourceObject, fieldNames = []) => {
  const target = {};

  for (const [key, value] of Object.entries(sourceObject)) {
    if (!isObject(value)) {
      target[key] = value;
      continue;
    }

    if (fieldNames.length > 0 && !fieldNames.includes(key)) {
      target[key] = { ...value };
      continue;
    }

    for (const [childKey, childValue] of Object.entries(value)) {
      const combinedKey = `${key}${toUpperCamel(childKey)}`;

      if (sourceObject[combinedKey])
        throw new Error(
          `Can not combine keys: key "${combinedKey}" already exists`,
        );

      target[combinedKey] = childValue;
    }
  }

  return target;
};
```